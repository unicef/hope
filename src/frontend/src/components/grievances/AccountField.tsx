import { Grid2 as Grid, IconButton, Button } from '@mui/material';
import { Delete, Add } from '@mui/icons-material';
import { useLocation } from 'react-router-dom';
import { Field, FieldArray } from 'formik';
import { useTranslation } from 'react-i18next';
import { FormikTextField } from '@shared/Formik/FormikTextField';
import { LabelizedField } from '@core/LabelizedField';
import { getIndexForId } from './utils/helpers';
import React, { Fragment, ReactElement } from 'react';
import { FormikSelectField } from '@shared/Formik/FormikSelectField';
import { Account } from '@restgenerated/models/Account';


export interface AccountProps {
  id: string;
  baseName: string;
  onDelete;
  isEdited?: boolean;
  account?: Account;
  values;
  accountTypeChoices: Record<string, any>[];
  accountFinancialInstitutionChoices: Record<string, any>[];
}

export function AccountField({
                               id,
                               baseName,
                               onDelete,
                               isEdited,
                               account,
                               values,
                               accountTypeChoices,
                               accountFinancialInstitutionChoices,
                             }: AccountProps): ReactElement {
  const { t } = useTranslation();

  const accountFieldName = `${baseName}.${getIndexForId(
    values[baseName],
    id,
  )}`;

  const location = useLocation();
  const isEditTicket = location.pathname.indexOf('edit-ticket') !== -1;
  const dataFields = account?.dataFields || {};

  const dynamicFieldsName = `${accountFieldName}.dynamicFields`;

  return (
    <>
      <Grid size={{ xs: 11 }}/>
      {!isEdited ? (
        <Grid size={{ xs: 1 }}>
          <IconButton disabled={isEditTicket} onClick={onDelete}>
            <Delete/>
          </IconButton>
        </Grid>
      ) : null}

      <Fragment key="type">
        <Grid size={{ xs: 4 }}>
          <LabelizedField
            label={t('Account Item')}
            value="type"
          />
        </Grid>

        {account ? (
          <>
            <Grid size={{ xs: 4 }}>
              <LabelizedField
                label={t('Current Value')}
                value={account?.accountType || ''}
              />
            </Grid>
            <Grid size={{ xs: 3 }}>
              <LabelizedField
                label={t('New Value')}
                value={account?.accountType || ''}
              />
            </Grid>
          </>
        ) : (
          <Grid size={{ xs: 8 }}>
            <Field
              name={`${accountFieldName}.name`}
              variant="outlined"
              label={t('New Value')}
              component={FormikSelectField}
              choices={accountTypeChoices}
              required
            />
          </Grid>
        )}
      </Fragment>

      {!account ? (
        <>
          <Fragment key="number">
            <Grid size={{ xs: 4 }}>
              <LabelizedField
                label={t('Account Item')}
                value="number"
              />
            </Grid>
            <Grid size={{ xs: 8 }}>
              <Field
                fullWidth
                name={`${accountFieldName}.number`}
                variant="outlined"
                label={t('New Value')}
                component={FormikTextField}
                required
              />
            </Grid>
          </Fragment>
          <Fragment key="financial_institution">
            <Grid size={{ xs: 4 }}>
              <LabelizedField
                label={t('Account Item')}
                value="financial_institution"
              />
            </Grid>
            <Grid size={{ xs: 8 }}>
              <Field
                name={`${accountFieldName}.financial_institution`}
                variant="outlined"
                label={t('New Value')}
                component={FormikSelectField}
                choices={accountFinancialInstitutionChoices}
                required
              />
            </Grid>
          </Fragment>
        </>
      ) : (
        <>
          {Object.entries(dataFields).map(([key, value]) => {
            let displayValue = String(value);
              const isFinancialInstitutionField = key === 'financial_institution';
              if (
                isFinancialInstitutionField &&
                Array.isArray(accountFinancialInstitutionChoices)
              ) {
                const choice = accountFinancialInstitutionChoices.find(
                  (c: any) => c.value === value,
                );
                displayValue = choice ? choice.name : String(value);
              }
              const fieldProps = {
                component: isFinancialInstitutionField ? FormikSelectField : FormikTextField,
                ...(isFinancialInstitutionField ? { choices: accountFinancialInstitutionChoices } : {}),
              };

            return (
            <Fragment key={key}>
              <Grid size={{ xs: 4 }}>
                <LabelizedField
                  label={t('Account Item')}
                  value={String(key)}
                />
              </Grid>
              <Grid size={{ xs: 4 }}>
                <LabelizedField
                  label={t('Current Value')}
                  value={displayValue}
                />
              </Grid>
              <Grid size={{ xs: 3 }}>
                  <Field
                      name={`${accountFieldName}.${key}`}
                      fullWidth
                      variant="outlined"
                      label={t('New Value')}
                      required={!isEditTicket}
                      disabled={isEditTicket}
                      {...fieldProps}
                    />
              </Grid>
            </Fragment>
          );
})}
        </>
      )
      }
      {/* --- Dynamic Fields Section --- */}
      <FieldArray name={dynamicFieldsName}>
        {/* eslint-disable-next-line @typescript-eslint/unbound-method*/}
        {({ push, remove, form }) => (
          <>
            {form.values[baseName][getIndexForId(values[baseName], id)]?.dynamicFields?.map(
              (field, idx) => (
                <Fragment key={idx}>
                  <Grid size={{ xs: 4 }}>
                    <Field
                      name={`${dynamicFieldsName}.${idx}.key`}
                      component={FormikTextField}
                      label={t('Field Name')}
                      variant="outlined"
                      fullWidth
                      required
                    />
                  </Grid>
                  <Grid size={{ xs: 7 }}>
                    <Field
                      name={`${dynamicFieldsName}.${idx}.value`}
                      component={FormikTextField}
                      label={t('Field Value')}
                      variant="outlined"
                      fullWidth
                      required
                    />
                  </Grid>
                  <Grid size={{ xs: 1 }}>
                    <IconButton onClick={() => remove(idx)}>
                      <Delete/>
                    </IconButton>
                  </Grid>
                </Fragment>
              ),
            )}
            <Grid size={{ xs: 12 }}>
              <Button
                variant="outlined"
                startIcon={<Add/>}
                onClick={() => push({ key: '', value: '' })}
              >
                {t('Add Field')}
              </Button>
            </Grid>
          </>
        )}
      </FieldArray>
      {/* --- End Dynamic Fields --- */}
    </>
  );
}