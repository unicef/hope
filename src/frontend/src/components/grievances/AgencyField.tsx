import { Grid, IconButton } from '@mui/material';
import { Delete } from '@mui/icons-material';
import { useLocation } from 'react-router-dom';
import { Field } from 'formik';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { FormikSelectField } from '@shared/Formik/FormikSelectField';
import { FormikTextField } from '@shared/Formik/FormikTextField';
import { AllAddIndividualFieldsQuery } from '@generated/graphql';
import { getIndexForId } from './utils/helpers';

export interface AgencyFieldProps {
  id: string;
  baseName: string;
  baseNameArray?;
  onDelete;
  countryChoices: AllAddIndividualFieldsQuery['countriesChoices'];
  identityTypeChoices: AllAddIndividualFieldsQuery['identityTypeChoices'];
  isEdited?: boolean;
  values;
}

export function AgencyField({
  id,
  baseName,
  baseNameArray,
  onDelete,
  countryChoices,
  identityTypeChoices,
  isEdited,
  values,
}: AgencyFieldProps): React.ReactElement {
  const { t } = useTranslation();
  const agencyFieldName = `${baseName}.${getIndexForId(
    baseNameArray || values[baseName],
    id,
  )}`;

  const location = useLocation();
  const isEditTicket = location.pathname.indexOf('edit-ticket') !== -1;
  return (
    <Grid container alignItems="center" spacing={3}>
      <Grid item xs={4}>
        <Field
          name={`${agencyFieldName}.partner`}
          fullWidth
          variant="outlined"
          label={t('Partner')}
          component={FormikSelectField}
          choices={identityTypeChoices}
          required
          disabled={isEditTicket}
        />
      </Grid>
      <Grid item xs={4}>
        <Field
          name={`${agencyFieldName}.country`}
          fullWidth
          variant="outlined"
          label={t('Country')}
          component={FormikSelectField}
          choices={countryChoices}
          required
          disabled={isEditTicket}
        />
      </Grid>
      <Grid item xs={3}>
        <Field
          name={`${agencyFieldName}.number`}
          fullWidth
          variant="outlined"
          label={t('Identity Number')}
          component={FormikTextField}
          required
          disabled={isEditTicket}
        />
      </Grid>
      {!isEdited ? (
        <Grid item xs={1}>
          <IconButton disabled={isEditTicket} onClick={onDelete}>
            <Delete />
          </IconButton>
        </Grid>
      ) : null}
    </Grid>
  );
}
