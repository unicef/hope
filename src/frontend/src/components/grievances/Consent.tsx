import * as React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

const ConsentText = styled.p`
  font-size: 14px;
  color: #585858;
`;

export function Consent(): React.ReactElement {
  const { t } = useTranslation();
  return (
    <ConsentText>
      {t(
        'Do you give your consent to UNICEF and its partners to view, edit and update your personal details and, if applicable, that of your household and dependants the purpose of the integrity UNICEFs beneficiary management system? Do you declare that the information you have provided is true and correct to the best of your knowledge?',
      )}
    </ConsentText>
  );
}
