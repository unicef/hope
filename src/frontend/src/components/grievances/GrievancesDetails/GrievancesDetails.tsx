import { Box, Grid, GridSize, Typography } from '@mui/material';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import {
  GrievanceTicketQuery,
  GrievancesChoiceDataQuery,
} from '@generated/graphql';
import { GRIEVANCE_CATEGORIES, GRIEVANCE_ISSUE_TYPES } from '@utils/constants';
import {
  choicesToDict,
  grievanceTicketBadgeColors,
  grievanceTicketStatusToColor,
  renderUserName,
} from '@utils/utils';
import { ContainerColumnWithBorder } from '@core/ContainerColumnWithBorder';
import { ContentLink } from '@core/ContentLink';
import { LabelizedField } from '@core/LabelizedField';
import { OverviewContainer } from '@core/OverviewContainer';
import { PhotoModal } from '@core/PhotoModal/PhotoModal';
import { StatusBox } from '@core/StatusBox';
import { Title } from '@core/Title';
import { UniversalMoment } from '@core/UniversalMoment';
import { useBaseUrl } from '@hooks/useBaseUrl';
import { BlackLink } from '@core/BlackLink';

interface GrievancesDetailsProps {
  ticket: GrievanceTicketQuery['grievanceTicket'];
  choicesData: GrievancesChoiceDataQuery;
  baseUrl: string;
  canViewHouseholdDetails: boolean;
  canViewIndividualDetails: boolean;
}

export function GrievancesDetails({
  ticket,
  choicesData,
  baseUrl,
  canViewHouseholdDetails,
  canViewIndividualDetails,
}: GrievancesDetailsProps): React.ReactElement {
  const { t } = useTranslation();
  const { isAllPrograms } = useBaseUrl();
  const statusChoices: {
    [id: number]: string;
  } = choicesToDict(choicesData.grievanceTicketStatusChoices);

  const priorityChoicesData = choicesData.grievanceTicketPriorityChoices;
  const urgencyChoicesData = choicesData.grievanceTicketUrgencyChoices;

  const categoryChoices: {
    [id: number]: string;
  } = choicesToDict(choicesData.grievanceTicketCategoryChoices);

  const issueType = ticket.issueType
    ? choicesData.grievanceTicketIssueTypeChoices
        .filter((el) => el.category === ticket.category.toString())[0]
        .subCategories.filter(
          (el) => el.value === ticket.issueType.toString(),
        )[0].name
    : '-';

  const showIssueType =
    ticket.category.toString() ===
      GRIEVANCE_CATEGORIES.SENSITIVE_GRIEVANCE.toString() ||
    ticket.category.toString() ===
      GRIEVANCE_CATEGORIES.DATA_CHANGE.toString() ||
    ticket.category.toString() ===
      GRIEVANCE_CATEGORIES.GRIEVANCE_COMPLAINT.toString() ||
    ticket.category.toString() ===
      GRIEVANCE_CATEGORIES.NEEDS_ADJUDICATION.toString();
  const showPartner =
    ticket.issueType === +GRIEVANCE_ISSUE_TYPES.PARTNER_COMPLAINT;

  const mappedDocumentation = (): React.ReactElement => (
    <Box display="flex" flexDirection="column">
      {ticket.documentation?.length
        ? ticket.documentation.map((doc) => {
            if (doc.contentType.includes('image')) {
              return (
                <PhotoModal
                  key={doc.id}
                  src={doc.filePath}
                  variant="link"
                  linkText={doc.name}
                />
              );
            }
            return (
              <ContentLink key={doc.id} download href={doc.filePath}>
                {doc.name}
              </ContentLink>
            );
          })
        : '-'}
    </Box>
  );

  const renderUrl = (
    obj,
    objType: string,
    href: string,
    displayedId: string,
  ): React.ReactElement => {
    if (isAllPrograms) {
      return <>{displayedId}</>;
    }

    if (obj?.objType === objType) {
      return <ContentLink href={href}>{displayedId}</ContentLink>;
    }

    return <>-</>;
  };

  const getUrl = (objType: string, id: string): string => {
    switch (objType) {
      case 'PaymentRecord':
        return `/${baseUrl}/payment-records/${id}`;
      case 'Payment':
        return `/${baseUrl}/payment-module/payments/${id}`;
      case 'PaymentPlan':
        return `/${baseUrl}/payment-module/payment-plans/${id}`;
      case 'CashPlan':
        return `/${baseUrl}/cashplans/${id}`;
      default:
        return '';
    }
  };

  const renderPaymentUrl = (): React.ReactElement => {
    const paymentRecord = ticket?.paymentRecord;
    if (paymentRecord) {
      return renderUrl(
        paymentRecord,
        paymentRecord.objType,
        getUrl(paymentRecord.objType, paymentRecord.id),
        paymentRecord.caId,
      );
    }
    return <>-</>;
  };

  const renderPaymentPlanUrl = (): React.ReactElement => {
    const parent = ticket?.paymentRecord?.parent;
    if (parent) {
      return renderUrl(
        parent,
        parent.objType,
        getUrl(parent.objType, parent.id),
        parent.unicefId,
      );
    }
    return <>-</>;
  };

  const renderPaymentPlanVerificationUrl = (): React.ReactElement => {
    const parent = ticket?.paymentRecord?.parent;
    if (parent) {
      const url = `/${baseUrl}/payment-verification/${
        parent.objType === 'CashPlan' ? 'cash-plan' : 'payment-plan'
      }/${parent.id}`;
      return renderUrl(parent, parent.objType, url, parent.unicefId);
    }
    return <>-</>;
  };

  const mappedPrograms = (): React.ReactElement => {
    if (!ticket.programs?.length) {
      return <>-</>;
    }
    return (
      <Box display="flex" flexDirection="column">
        {ticket.programs.map((program) => (
          <ContentLink
            key={program.id}
            href={`/${baseUrl}/details/${program.id}`}
          >
            {program.name}
          </ContentLink>
        ))}
      </Box>
    );
  };

  return (
    <Grid item xs={12}>
      <ContainerColumnWithBorder>
        <Title>
          <Typography variant="h6">{t('Details')}</Typography>
        </Title>
        <OverviewContainer>
          <Grid container spacing={6}>
            {[
              {
                label: t('Status'),
                value: (
                  <StatusBox
                    status={statusChoices[ticket.status]}
                    statusToColor={grievanceTicketStatusToColor}
                  />
                ),
                size: 3,
              },
              {
                label: t('Priority'),
                value: (
                  <StatusBox
                    status={
                      priorityChoicesData[
                        priorityChoicesData.findIndex(
                          (obj) => obj.value === ticket.priority,
                        )
                      ]?.name || '-'
                    }
                    statusToColor={grievanceTicketBadgeColors}
                  />
                ),
                size: 3,
              },
              {
                label: t('Urgency'),
                value: (
                  <StatusBox
                    status={
                      urgencyChoicesData[
                        urgencyChoicesData.findIndex(
                          (obj) => obj.value === ticket.urgency,
                        )
                      ]?.name || '-'
                    }
                    statusToColor={grievanceTicketBadgeColors}
                  />
                ),
                size: 3,
              },
              {
                label: t('Assigned to'),
                value: renderUserName(ticket.assignedTo),
                size: 3,
              },
              {
                label: t('Category'),
                value: <span>{categoryChoices[ticket.category]}</span>,
                size: 3,
              },
              showIssueType && {
                label: t('Issue Type'),
                value: <span>{issueType}</span>,
                size: 3,
              },
              {
                label: t('Household ID'),
                value: (
                  <span>
                    {ticket.household?.id &&
                    canViewHouseholdDetails &&
                    !isAllPrograms ? (
                      <BlackLink
                        to={`/${baseUrl}/population/household/${ticket.household.id}`}
                      >
                        {ticket.household.unicefId}
                      </BlackLink>
                    ) : (
                      <div>
                        {ticket.household?.id ? ticket.household.unicefId : '-'}
                      </div>
                    )}
                  </span>
                ),
                size: 3,
              },
              {
                label: t('Individual ID'),
                value: (
                  <span>
                    {ticket.individual?.id &&
                    canViewIndividualDetails &&
                    !isAllPrograms ? (
                      <BlackLink
                        to={`/${baseUrl}/population/individuals/${ticket.individual.id}`}
                      >
                        {ticket.individual.unicefId}
                      </BlackLink>
                    ) : (
                      <div>
                        {ticket.individual?.id
                          ? ticket.individual.unicefId
                          : '-'}
                      </div>
                    )}
                  </span>
                ),
                size: 3,
              },
              {
                label: t('Payment ID'),
                value: <span>{renderPaymentUrl()}</span>,
                size: 3,
              },
              {
                label: t('Payment Plan'),
                value: <span>{renderPaymentPlanUrl()}</span>,
                size: 3,
              },
              {
                label: t('Payment Plan Verification'),
                value: <span>{renderPaymentPlanVerificationUrl()}</span>,
                size: 3,
              },
              {
                label: t('Programme'),
                value: mappedPrograms(),
                size: 3,
              },
              showPartner && {
                label: t('Partner'),
                value: ticket.partner?.name,
                size: 3,
              },
              {
                label: t('Created By'),
                value: renderUserName(ticket.createdBy),
                size: 3,
              },
              {
                label: t('Date Created'),
                value: <UniversalMoment>{ticket.createdAt}</UniversalMoment>,
                size: 3,
              },
              {
                label: t('Last Modified Date'),
                value: <UniversalMoment>{ticket.updatedAt}</UniversalMoment>,
                size: 3,
              },
              {
                label: t('Administrative Level 2'),
                value: ticket.admin,
                size: 3,
              },
              {
                label: t('Area / Village / Pay point'),
                value: ticket.area,
                size: 3,
              },
              {
                label: t('Languages Spoken'),
                value: ticket.language,
                size: 3,
              },
              {
                label: t('Documentation'),
                value: mappedDocumentation(),
                size: 3,
              },
              {
                label: t('Description'),
                value: ticket.description,
                size: 12,
              },
              {
                label: t('Comments'),
                value: ticket.comments,
                size: 12,
              },
            ]
              .filter((el) => el)
              .map((el) => (
                <Grid key={el.label} item xs={el.size as GridSize}>
                  <LabelizedField label={el.label}>{el.value}</LabelizedField>
                </Grid>
              ))}
          </Grid>
        </OverviewContainer>
      </ContainerColumnWithBorder>
    </Grid>
  );
}
