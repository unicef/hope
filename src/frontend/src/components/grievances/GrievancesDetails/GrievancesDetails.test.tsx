import * as React from 'react';
import { render } from '../../../testUtils/testUtils';
import { fakeGrievanceTicket } from '../../../../fixtures/grievances/fakeGrievanceTicket';
import { fakeGrievancesChoices } from '../../../../fixtures/grievances/fakeGrievancesChoices';
import { fakeBaseUrl } from '../../../../fixtures/core/fakeBaseUrl';
import { GrievancesDetails } from './GrievancesDetails';

describe('components/grievances/GrievancesDetails', () => {
  it('should render', () => {
    const { container } = render(
      <GrievancesDetails
        baseUrl={fakeBaseUrl}
        ticket={fakeGrievanceTicket}
        choicesData={fakeGrievancesChoices}
        canViewHouseholdDetails
        canViewIndividualDetails
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
