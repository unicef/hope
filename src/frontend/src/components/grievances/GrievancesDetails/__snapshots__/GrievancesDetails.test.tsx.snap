// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grievances/GrievancesDetails should render 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-blHHSb kPa-dfK css-1ps6pg7-MuiPaper-root"
    >
      <div
        class="sc-dpBQxM jHeYpq"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1j53xw6-MuiTypography-root"
        >
          Details
        </h6>
      </div>
      <div
        class="sc-dntaoT fMVexb"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 css-zow5z4-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Status
              </span>
              <div
                data-cy="label-Status"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <div
                    class="sc-dstKZu jHyLSU"
                  >
                    <div
                      class="sc-keTIit gWjvwv status-box-container"
                      data-cy="status-container"
                    >
                      In Progress
                    </div>
                  </div>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Priority
              </span>
              <div
                data-cy="label-Priority"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <div
                    class="sc-dstKZu jHyLSU"
                  >
                    <div
                      class="sc-keTIit hA-dCPj status-box-container"
                      data-cy="status-container"
                    >
                      High
                    </div>
                  </div>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Urgency
              </span>
              <div
                data-cy="label-Urgency"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <div
                    class="sc-dstKZu jHyLSU"
                  >
                    <div
                      class="sc-keTIit hA-dCPj status-box-container"
                      data-cy="status-container"
                    >
                      Very urgent
                    </div>
                  </div>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Assigned to
              </span>
              <div
                data-cy="label-Assigned to"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  Maciej Szewczyk
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Category
              </span>
              <div
                data-cy="label-Category"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <span>
                    Data Change
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Issue Type
              </span>
              <div
                data-cy="label-Issue Type"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <span>
                    Withdraw Individual
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Household ID
              </span>
              <div
                data-cy="label-Household ID"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <span>
                    <a
                      class="sc-jwIPbr gLUwzC"
                      href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/population/household/SG91c2Vob2xkTm9kZTo3NjExNzM2Ny0yYWFiLTRmNTEtODUwOC1mMzBmODliYWUzYzE="
                    >
                      HH-20-0000.0001
                    </a>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Individual ID
              </span>
              <div
                data-cy="label-Individual ID"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <span>
                    <a
                      class="sc-jwIPbr gLUwzC"
                      href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/population/individuals/SW5kaXZpZHVhbE5vZGU6ZjVmMzgwN2ItYTBjOS00Mjk0LTg0Y2QtNDhkMjllMjZhODU3"
                    >
                      IND-88-0000.0005
                    </a>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Payment ID
              </span>
              <div
                data-cy="label-Payment ID"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <span>
                    -
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Payment Plan
              </span>
              <div
                data-cy="label-Payment Plan"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <span>
                    -
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Payment Plan Verification
              </span>
              <div
                data-cy="label-Payment Plan Verification"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <span>
                    -
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Programme
              </span>
              <div
                data-cy="label-Programme"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  -
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Created By
              </span>
              <div
                data-cy="label-Created By"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  Maciej Szewczyk
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Date Created
              </span>
              <div
                data-cy="label-Date Created"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <time
                    datetime="1644571534682"
                  >
                    11 Feb 2022
                  </time>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Last Modified Date
              </span>
              <div
                data-cy="label-Last Modified Date"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <time
                    datetime="1644571964102"
                  >
                    11 Feb 2022
                  </time>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Administrative Level 2
              </span>
              <div
                data-cy="label-Administrative Level 2"
              >
                -
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Area / Village / Pay point
              </span>
              <div
                data-cy="label-Area / Village / Pay point"
              >
                -
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Languages Spoken
              </span>
              <div
                data-cy="label-Languages Spoken"
              >
                -
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Documentation
              </span>
              <div
                data-cy="label-Documentation"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  <div
                    class="MuiBox-root css-j7qwjs"
                  >
                    -
                  </div>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Description
              </span>
              <div
                data-cy="label-Description"
              >
                <span
                  class="sc-fAUdSK fuvdWP"
                  color="textSecondary"
                >
                  dwdw
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div>
              <span
                class="sc-egkSDF bUfwNm"
                color="textSecondary"
              >
                Comments
              </span>
              <div
                data-cy="label-Comments"
              >
                -
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
